lib/dynamo/dynamo/jobs.py

```diff
@@ -117,22 +117,35 @@ def _prepare_job_for_database(
 
 def _get_credit_cost(job: dict, costs: list[dict]) -> Decimal:
     job_type = job['job_type']
-    for cost_definition in costs:
-        if cost_definition['job_type'] == job_type:
-            if cost_definition.keys() not in ({'job_type', 'cost_parameter', 'cost_table'}, {'job_type', 'cost'}):
-                raise ValueError(f'Cost definition for job type {job_type} has invalid keys: {cost_definition.keys()}')
-
-            if 'cost_parameter' in cost_definition:
-                cost_parameter = cost_definition['cost_parameter']
-                parameter_value = job['job_parameters'][cost_parameter]
-                for item in cost_definition['cost_table']:
-                    if item['parameter_value'] == parameter_value:
-                        return item['cost']
-                raise ValueError(f'Cost not found for job type {job_type} with {cost_parameter} == {parameter_value}')
-
-            return cost_definition['cost']
-    raise ValueError(f'Cost not found for job type {job_type}')
+    cost_definition = costs[job_type]
 
+    if cost_definition.keys() not in ({'cost_parameters', 'cost_table'}, {'cost'}):
+        raise ValueError(f'Cost definition for job type {job_type} has invalid keys: {cost_definition.keys()}')
+
+    if 'cost' in cost_definition:
+        return cost_definition['cost']
+    else:
+        return _get_cost_from_table(job, cost_definition)
+
+
+def _get_cost_from_table(job, cost_definition):
+    cost_table = cost_definition['cost_table']
+
+    for cost_parameter in cost_definition['cost_parameters']:
+        parameter_value = job['job_parameters'][cost_parameter]
+
+        if isinstance(parameter_value, list):
+            parameter_value = len(parameter_value)
+
+        if isinstance(parameter_value, float):
+            parameter_value = int(parameter_value)
```

Comment:
Oh OK, I see you're already handling the float -> int conversion. We probably want to verify that `parameter_value == int(parameter_value)` to make sure the float value doesn't have a fractional part. Though, we may also want to consider just changing how the resolution parameter works for `RTC_GAMMA`. Ideally I'd probably just convert it to an integer in the API spec, but I think that would break backwards compatibility. Something to think about more later.

Reply:
Nevermind, I was overcomplicating this. What you have here is probably good.